Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderView.kt

```diff
@@ -43,6 +44,13 @@ fun ApproveOrderView(
         // continuously scroll to bottom of the list when scroll bounds change
         scrollState.animateScrollTo(scrollState.maxValue)
     }
+
+    val context = LocalContext.current
+    OnLifecycleOwnerResumeEffect {
+        val intent = context.getActivityOrNull()?.intent
+        intent?.let { viewModel.completeAuthChallenge(intent) }
```

Comment:
This is taking place of previous CardLifeCycleObserver, right?

Reply:
Yes `CardLifeCycleObserver` used to do something similar behind the scenes. The `v2` integration requires the merchant to explicitly complete the auth challenge.

Reply:
Let me summarize what I understand so far:
- There was change from passing in AppCompatActivity to ComponentActivity into approveOrder function
- Lifecycle Observer was removed from SDK
- Merchant application is now responsible for managing lifecycle and presenting authChallenge.

These changes were made for edge cases where there may be memory leaks in different merchant app implementations and undefined behavior in cases of process kill

Reply:
Yup! That's it in summary. There will be follow up PRs to remove listeners as well. Ideally the SDK will be completely stateless in `v2`.

Reply:
Nice work Steven! This is a lot!

Reply:
🙏 hopefully will be less going forward! This PR is a big tech debt payment. I'm glad we're able to get it merged.